app/assets/js/components/Work/Tabs/Preservation/S3ObjectPicker.jsx

```diff
@@ -0,0 +1,115 @@
+import useAcceptedMimeTypes from "@js/hooks/useAcceptedMimeTypes";
+import { Button } from "@nulib/design-system";
+import {
+  LIST_INGEST_BUCKET_OBJECTS,
+} from "@js/components/Work/work.gql.js";
+import React, { useState } from "react";
+/** @jsx jsx */
+import { css, jsx } from "@emotion/react";
+import { useQuery } from "@apollo/client";
+import { FaSpinner } from "react-icons/fa";
+import { formatBytes } from "@js/services/helpers";
+
+import Error from "@js/components/UI/Error";
+import UIFormInput from "@js/components/UI/Form/Input.jsx";
+
+const tableContainerCss = css`
+  max-height: 30vh;
+  overflow-y: auto;
+`;
+
+const fileRowCss = css`
+  cursor: pointer;
+`;
+
+// a nice gentle blue
+const selectedRowCss = css`
+  background-color: #f0f8ff !important;
+`;
+
+const colHeaders = ["File Key", "Size", "Mime Type"];
+
+const { isFileValid } = useAcceptedMimeTypes();
+
+const S3ObjectPicker = ({ onFileSelect, fileSetRole, workTypeId, defaultPrefix = "" }) => {
+  const [prefix, setPrefix] = useState(defaultPrefix);
+  const [selectedFile, setSelectedFile] = useState(null);
+  const [error, _setError] = useState(null);
+
+  const { loading: queryLoading, error: queryError, data, refetch } = useQuery(LIST_INGEST_BUCKET_OBJECTS, {
+    variables: { prefix }
+  });
+
+  const handleClear = () => {
+    setPrefix(defaultPrefix);
+    refetch({ prefix: defaultPrefix });
+  };
+
+  const handlePrefixChange = (e) => {
+    const inputValue = e.target.value;
+    const newPrefix = inputValue.startsWith(defaultPrefix) ? inputValue : defaultPrefix + inputValue;
+    setPrefix(newPrefix);
+    refetch({ prefix: newPrefix });
```

Comment:
This `refetch()` is `async` also?  Works ok w/o the `await`?

Reply:
I didn't think to async await the call to refresh here, but I can give it a shot.

Reply:
@adamjarling Just pushed a commit to address this, `await` working in my dev env 👍